doc/contributing/technical-priorities.md

```diff
@@ -160,3 +160,18 @@ and integration with other systems.
 As TypeScript usage continues to grow and gains more prominence in the
 ecosystem, enhancing its support is essential for delivering an improved
 developer experience for newcomers and experienced users alike.
+
+## Non-goals
+
+The following are not considered technical priorities for the project:
+
+* **Unbundling `npm`**. When `npm` was included in the Node.js distribution, it
+  was the only JavaScript package manager available, and it was provided as a
+  way to help developers easily install other JavaScript software. `npm` is also
+  the reference implementation for the `npm` registry, which is the de facto
+  primary source for most JavaScript software. Today, `npm` is one of many
+  high-quality options. However, the potential removal of `npm` would be a very
+  disruptive breaking change, even as a semver-major change, and therefore it is
+  not a goal of the Node.js project to work toward such a goal. The `npm`
+  library included with the Node.js distribution can be freely used as a
+  dependency of other parts of the Node.js distribution.
```

Comment:
We don't expose a library, do we?
```suggestion
  not a goal of the Node.js project to work toward such a goal.
```

Reply:
If we implement https://github.com/nodejs/node/issues/51931, the most straightforward solution that I can think of would be use the npm that’s provided as part of the distribution. Like for example, when you run `corepack enable`, it creates `/usr/local/bin/yarn` that’s a symlink to `../lib/node_modules/corepack/dist/yarn.js` and contains:

```js
#!/usr/bin/env node
require('./corepack').runMain(['yarn', ...process.argv.slice(2)]);
```

To implement https://github.com/nodejs/node/issues/51931, one way would be to have `/usr/local/bin/yarn` be a script with something like `require('../lib/node_modules/npm/bin/npm-cli.js')` to access npm and then use it to install Corepack and/or Yarn/pnpm.

An objection to https://github.com/nodejs/node/issues/51931 was that this potential implementation creates a dependency on npm. I think that should not be a valid objection, since we’re not planning on unbundling npm. And of course, nothing is permanent; if we land https://github.com/nodejs/node/issues/51931 and afterward decide we want to unbundle npm, then as part of removing npm we would need to reimplement https://github.com/nodejs/node/issues/51931 to not depend on npm. But I think that’s fine, since it seems unlikely that we would want to unbundle npm anytime soon.